lxd/main_init.go

```diff
@@ -273,44 +281,256 @@ they otherwise would.
+	// in case it's needed later.
+	return reverter, nil
+}
+
+// Create or update a single pool, and return a revert function in case of success.
+func (cmd *CmdInit) initPool(client lxd.ContainerServer, pool api.StoragePoolsPost) (reverter, error) {
+	var reverter func() error
+	currentPool, _, err := client.GetStoragePool(pool.Name)
+	if err == nil {
+		reverter, err = cmd.initPoolUpdate(client, pool, currentPool.Writable())
+	} else {
+		reverter, err = cmd.initPoolCreate(client, pool)
+	}
+	if err != nil {
+		return nil, err
+	}
+	return reverter, nil
+}
+
+// Create a single new pool, and return a revert function to delete it.
+func (cmd *CmdInit) initPoolCreate(client lxd.ContainerServer, pool api.StoragePoolsPost) (reverter, error) {
+	reverter := func() error {
+		return client.DeleteStoragePool(pool.Name)
+	}
+	err := client.CreateStoragePool(pool)
+	return reverter, err
+}
+
+// Update a single pool, and return a function that can be used to
+// revert it to its original state.
+func (cmd *CmdInit) initPoolUpdate(client lxd.ContainerServer, pool api.StoragePoolsPost, currentPool api.StoragePoolPut) (reverter, error) {
+	reverter := func() error {
+		return client.UpdateStoragePool(pool.Name, currentPool, "")
+	}
+	err := client.UpdateStoragePool(pool.Name, api.StoragePoolPut{
+		Config: pool.Config,
+	}, "")
+	return reverter, err
+}
+
+// Create or update a single network, and return a revert function in case of success.
+func (cmd *CmdInit) initNetwork(client lxd.ContainerServer, network api.NetworksPost) (reverter, error) {
+	var revert func() error
+	currentNetwork, _, err := client.GetNetwork(network.Name)
+	if err == nil {
+		// Sanity check, make sure the network type being updated
+		// is still "bridge", which is the only type the existing
+		// network can have.
+		if network.Type != "" && network.Type != "bridge" {
+			return nil, fmt.Errorf("Only 'bridge' type networks are supported")
+		}
+		revert, err = cmd.initNetworkUpdate(client, network, currentNetwork.Writable())
+	} else {
+		revert, err = cmd.initNetworkCreate(client, network)
+	}
+	if err != nil {
+		return nil, err
+	}
+	return revert, nil
+}
+
+// Create a single new network, and return a revert function to delete it.
+func (cmd *CmdInit) initNetworkCreate(client lxd.ContainerServer, network api.NetworksPost) (reverter, error) {
+	reverter := func() error {
+		return client.DeleteNetwork(network.Name)
+	}
+	err := client.CreateNetwork(network)
+	return reverter, err
+}
+
+// Update a single network, and return a function that can be used to
+// revert it to its original state.
+func (cmd *CmdInit) initNetworkUpdate(client lxd.ContainerServer, network api.NetworksPost, currentNetwork api.NetworkPut) (reverter, error) {
+	reverter := func() error {
+		return client.UpdateNetwork(network.Name, currentNetwork, "")
+	}
+	err := client.UpdateNetwork(network.Name, api.NetworkPut{
+		Config: network.Config,
+	}, "")
+	return reverter, err
+}
```

Comment:
Same comment as with storage, the network API doesn't exist in LXD 2.0.

Reply:
Same as above.

Reply:
Did you forget to remove that code too?

Reply:
Ah, a "don't do try to do that late at night commit". Dropped for real.